src/java/org/apache/cassandra/sensors/RequestSensors.java

```diff
@@ -110,6 +110,29 @@ public void syncAllSensors()
         }
     }
 
+    /**
+     * Avoids using additional ConcurrentHashmap as that turns out to be
+     * a performance bottlneck (in particular hashCode operations on Sensor were taking a lots of CPU cycles)
+     */
+    public void syncAllSensorsNew()
+    {
+        updateLock.writeLock().lock();
+        try
+        {
+            sensors.values().forEach(sensor -> {
+                if (sensor.getValue() > 0)
+                {
+                    sensorsRegistry.get().incrementSensor(sensor.getContext(), sensor.getType(), sensor.getValue());
+                    sensor.reset();
```

Comment:
NIT: maybe reset could be based on `getAndSet` to return the previous value, so something like:
```
sensorsRegistry.get().incrementSensor(sensor.getContext(), sensor.getType(), sensor.reset());
```

Reply:
Good point. After tweaking `syncAllSensorsNew` the throughput increased for about 6-7%:
```
Benchmark                              Mode  Cnt   Score   Error  Units
RequestSensorBench.syncAllSensorsNew  thrpt    5  75.061 ± 6.343  ops/s
```